worker/dbaccessor/worker.go

```diff
@@ -238,11 +271,64 @@ func (w *dbWorker) initializeDqlite() error {
 	return nil
 }
 
-type machAddressFactory func(string) network.MachineAddress
+func (w *dbWorker) getDQliteTLSConfiguration() (serverConf, clientConf *tls.Config, err error) {
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(w.cfg.ControllerCACert)
+
+	controllerCert, err := tls.X509KeyPair(w.cfg.ControllerCert, w.cfg.ControllerCertKey)
+	if err != nil {
+		return nil, nil, errors.Annotate(err, "parsing controller certificate")
+	}
+
+	return &tls.Config{
+			ClientCAs:    caCertPool,
+			Certificates: []tls.Certificate{controllerCert},
+		},
+		&tls.Config{
+			RootCAs:      caCertPool,
+			Certificates: []tls.Certificate{controllerCert},
+			// We cannot provide a ServerName value here so instead we
+			// rely on the server validating the controller's client cert.
+			InsecureSkipVerify: true,
+		},
+		nil
+}
+
+func detectLocalDQliteAddress() (string, error) {
+	sysAddrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return "", errors.Annotate(err, "querying addresses of system NICs")
+	}
+
+	// DQlite nodes should only advertise cloud-local addresses to their
+	// peers. To figure out the address to bind to, we need to scan all NIC
+	// addresses and return the first one that has cloud-local scope.
+	for _, sysAddr := range sysAddrs {
+		var host string
+
+		switch v := sysAddr.(type) {
+		case *net.IPNet:
+			host = v.IP.String()
+		case *net.IPAddr:
+			host = v.IP.String()
```

Comment:
default case?

Reply:
It's caught by the if block immediately below the switch.